File: src/main/java/net/arvandor/numinoustreasury/listener/InventoryCloseListener.java
package net.arvandor.numinoustreasury.listener;

import net.arvandor.numinoustreasury.item.NuminousItemInventoryHolder;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryCloseEvent;
import org.bukkit.inventory.Inventory;

public final class InventoryCloseListener implements Listener {

    @EventHandler
    public void onInventoryClose(InventoryCloseEvent event) {
        Inventory inventory = event.getInventory();
        if (!(inventory.getHolder() instanceof NuminousItemInventoryHolder holder)) {
            return;
        }
        holder.updateItemStack();
    }

}

File: src/main/kotlin/net/arvandor/numinoustreasury/listener/InventoryCloseListener.kt
package net.arvandor.numinoustreasury.listener

import net.arvandor.numinoustreasury.item.NuminousItemInventoryHolder
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryCloseEvent

class InventoryCloseListener : Listener {
    @EventHandler
    fun onInventoryClose(event: InventoryCloseEvent) {
        val inventory = event.inventory
        val holder = inventory.holder
        if (holder !is NuminousItemInventoryHolder) {
            return
        }
        holder.updateItemStack()
    }
}
